### PR TITLE
resource_control: improve trace logs, ctl and metrics

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7509

### What is changed and how does it work?

```commit-message
resource_control: improve trace logs, ctl and metrics
- add the ctl command to config resource cotnrol
- add metrics and a switch to enable trace log
```

### Check List

Tests

- Unit test
- Manual test (add detailed scripts or steps below)
image

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by nolouch** [comment]
ptal @CabinfeverB @bufferflies @glorv

**Review by bufferflies — APPROVED** [review]

File: client/resource_group/controller/controller.go [hunk 1]
```diff
@@ -1214,6 +1230,8 @@ func (gc *groupCostController) onRequestWait(
 				}
 				if d, err = WaitReservations(ctx, now, res); err == nil {
 					break retryLoop
+				} else if d.Seconds() > 0 {
+					gc.failedLimitReserveDuration.Observe(d.Seconds())
```
Comment (CabinfeverB) [inline client/resource_group/controller/controller.go:1234]: IMO, we don't need to record it when retry

**Review by CabinfeverB — COMMENTED** [review]
Rest LGTM

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**